lib/ActivityStreams.js

```diff
@@ -483,6 +489,23 @@ ActivityStreams.prototype = {
     });
   },
 
+  /*
+   * Replace the home page with the ActivityStream new tab page.
+   */
+  _setHomePage() {
+    // Only hijack the home page if it is set to about:home or about:blank
+    if (["about:home", "about:blank"].includes(prefService.get(HOME_PAGE_PREF))) {
+      prefService.set(HOME_PAGE_PREF, this._newTabURL);
+    }
+  },
+
+  _unsetHomePage() {
+    // Reset home page back if user didn't change it.
+    if (prefService.get(HOME_PAGE_PREF) === this._newTabURL) {
```

Comment:
This is where I conditionally reset it back. I think it is way simpler than adding listeners and all that jazz.